01_Data/src/layers/sequelize/repository/DeviceModel.ts

```diff
@@ -227,29 +275,29 @@ export class DeviceModelRepository extends SequelizeRepository<VariableAttribute
   async readAllByQuery(query: VariableAttributeQuerystring): Promise<VariableAttribute[]> {
     const readQuery = this.constructQuery(query);
     readQuery.include.push(VariableStatus);
-    return await super.readAllByQuery(readQuery, VariableAttribute.MODEL_NAME);
+    return await super.readAllByQuery(readQuery);
   }
 
-  async existsByQuery(query: VariableAttributeQuerystring): Promise<boolean> {
-    return await super.existsByQuery(this.constructQuery(query), VariableAttribute.MODEL_NAME);
+  async existByQuery(query: VariableAttributeQuerystring): Promise<number> {
+    return await super.existByQuery(this.constructQuery(query));
   }
 
-  async deleteAllByQuery(query: VariableAttributeQuerystring): Promise<number> {
-    return await super.deleteAllByQuery(this.constructQuery(query), VariableAttribute.MODEL_NAME);
+  async deleteAllByQuery(query: VariableAttributeQuerystring): Promise<VariableAttribute[]> {
+    return await super.deleteAllByQuery(this.constructQuery(query));
   }
 
   async findComponentAndVariable(componentType: ComponentType, variableType: VariableType): Promise<[Component | null, Variable | null]> {
-    const component = await Component.findOne({
+    const component = await this.component.readAllByQuery({
       where: { name: componentType.name, instance: componentType.instance ? componentType.instance : null },
-    });
-    const variable = await Variable.findOne({
+    }).then((components) => components.length > 0 ? components[0] : null);
```

Comment:
similar situation here where maybe we can benefit from having `findFirst` helper

Reply:
to eiliminate / centralize occurenses of logic like this 
```.then((variableCharacteristics) => variableCharacteristics.length > 0 ? variableCharacteristics[0] : undefined);
```
and have it in one place in the helpers / base

Reply:
I agree! Thanks for the suggestion; I've added a 'readOnlyOneByQuery' method which throws an error if more than one result is returned, and replaced situations like the above with a call to it.